instrumentation/instacosmos/container_client_test.go

```diff
@@ -168,6 +194,18 @@ func TestMain(m *testing.M) {
 	tracer := instana.NewTracerWithEverything(&instana.Options{AgentClient: alwaysReadyClient{}}, recorder)
 	sensor := instana.NewSensorWithTracer(tracer)
 
+	// handles panic errors
+	// this will only work for the panic errors before m.Run()
+	// check this issue for more details: https://github.com/golang/go/issues/37206
+	defer func() {
+		if err := recover(); err != nil {
+			shutdown(sensor, exitNotOk)
```

Comment:
Isn't it better to re-throw the panic after calling shutdown ?

Reply:
yes. Good Catch 👍